app/account/page.tsx

```diff
@@ -0,0 +1,29 @@
+"use client";
+
+import React from "react";
+import { isAuthenticated } from "@/firebase/authentication";
+import { redirect } from "next/navigation";
+import { getCurrentUser } from "@/firebase/authentication";
+import { SignOutButton } from "@/components/buttons";
+
+const ProfilePage: React.FC = () => {
+  if (!isAuthenticated()) {
+    redirect("/login");
```

Comment:
This works?

Reply:
I have redone it now as the "isAuthenticated" is sync but the check firebase makes is async. So yeah it worked but not optimally. Now i use the hook "useAuthState" which works smoothly